src/game/client/in_main.cpp

```diff
@@ -405,6 +405,51 @@ void KeyDown( kbutton_t *b, const char *c )
 	b->state |= 1 + 2;	// down + impulse down
 }
 
+#ifdef NEO
+/*
+============
+KeyDown where pressing a hold key will lift a toggle key
+============
+*/
+void KeyDownHoldReplaceToggle( kbutton_t *b, const char *c )
+{
+	int		k = -1;
+	if ( c && c[0] )
+	{
+		k = atoi(c);
+	}
+
+	if (k == b->down[0] || k == b->down[1])
+		return;		// repeating key
+	if (b->down[0] == -1)
+	{
+		b->down[0] = k;
+		return; // state hasn't changed
+	}
+	else if (b->down[1] == -1)
+	{
+		b->down[1] = k;
+		return; // state hasn't changed
+	}
+	else if (!b->down[0])
+		b->down[0] = k;
+	else if (!b->down[1])
+		b->down[1] = k;
+	else
+	{
+		if ( c[0] )
```

Comment:
For line 440: Since we're checking for nullptr at line 417, should we also do the same check here, to avoid a nullptr dereference? That being said, the same thing occurs at line 396 in Valve code. But maybe it'd be safe to add the nullptr check for both?